### PR TITLE
doc: added validation example with a processor

### PR DESCRIPTION
**Pull request by COil**
Hello,

* Should we tell how to add the processor tag when not using Symfony autoconfiguration ?

See you.

### CODE REVIEW

File: doc/advanced-usage.md [hunk 1]
```diff
@@ -236,5 +236,47 @@ services:
...
+    ) {
+    }
+
+    public function preProcess(string $id, object $object): void
+    {
+        /** @var ConstraintViolationList $violations */
+        $violations = $this->validator->validate($object);
+        if ($violations->count() > 0) {
+            $message = sprintf("Error when validating fixture %s, violation(s) detected:\n%s", $id, $violations);
+            throw new \DomainException($message);
```
Comment (theofidry) [inline doc/advanced-usage.md]: I would import this class as well

File: doc/advanced-usage.md [hunk 2]
```diff
@@ -236,5 +236,47 @@ services:
...
 
+## Validation
+
+By default, fixtures are not validated. If you want to trigger the validation you
+can do it with a processor:
+
+```php
+<?php
+
+declare(strict_types=1);
```
Comment (theofidry) [inline doc/advanced-usage.md]: I personally prefer this variant for code excepts

```suggestion
<?php declare(strict_types=1);
```

File: doc/advanced-usage.md [hunk 3]
```diff
@@ -236,5 +236,47 @@ services:
 Done.
 
 
+## Validation
+
+By default, fixtures are not validated. If you want to trigger the validation you
+can do it with a processor:
```
Comment (theofidry) [inline doc/advanced-usage.md]:
```suggestion
can do it with [a custom processor](alice-processors.md):
```

**Review by theofidry — APPROVED** [review]
I think it can be done like suggested, thanks for the doc entry it's a cool idea!

**Comment by COil** [comment]
Thanks for feedback Theo. 🙂

**Comment by theofidry** [comment]
Thank you @COil !

**Pull request merged**